optaweb-employee-rostering-backend/src/main/java/org/optaweb/employeerostering/domain/tenant/RosterConstraintConfiguration.java

```diff
@@ -30,24 +30,19 @@
 @ConstraintConfiguration(constraintPackage = "org.optaweb.employeerostering.service.solver")
 public class RosterConstraintConfiguration extends AbstractPersistable {
```

Comment:
Any chance you could reuse my ConstraintProvider PR here? It refactors this class to use String constants to describe constraints, and therefore reduces duplication and potential for human error.

Reply:
We might as well just merge the ConstraintProvider work - even if we don't enable it (and we shouldn't!), it doesn't hurt for the provider to be there. Mario and I are working on CS-D performance PoCs right now.

Reply:
I say merging the ConstraintProvider work is out of scope for this PR, but it could be done in a separate PR (although it might confuse people why there are two implementations. @ge0ffrey what do you think?

Reply:
The PR actually has two parts:
- One is the constraint provider,
- but the more important part is that is refactors the app to reuse constraint names across the app. Which, I would argue, is pretty important.

Reply:
The second point looks good; your PR has merge conflicts currently. I would prefer to merge it separately to keep the scope of this PR manageable.

Reply:
Merge conflicts solved, PR updated.
Please port at your earliest convenience.